src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java

```diff
@@ -377,6 +382,23 @@ private void buildAAA() {
             }
         }
 
+        /*
+         * If the Api token authentication is configured:
+         * Add the ApiToken authbackend in to the auth domains
+         * Challenge: false - no need to iterate through the auth domains again when ApiToken authentication failed
+         * order: -2 - prioritize the Api token authentication when it gets enabled
+         */
+        Settings apiTokenSettings = getDynamicApiTokenSettings();
+        if (!isKeyNull(apiTokenSettings, "signing_key")) {
+            final AuthDomain _ad = new AuthDomain(
+                new NoOpAuthenticationBackend(Settings.EMPTY, null),
+                new ApiTokenAuthenticator(getDynamicApiTokenSettings(), this.cih.getClusterName(), ar),
+                false,
+                -2
```

Comment:
If a user has configured an auth domain with the priority -2, this will be removed in favor of the `ApiTokenAuthenticator`. This might be unexpected behavior.